### PR TITLE
Host header: revert default, support preserveHost

### PR DESCRIPTION
**Pull request by aryszka**

### CODE REVIEW

File: proxy/doc.go [hunk 1]
```diff
@@ -95,6 +95,20 @@ from the external data sources, and are tested against the requests
...
+The default behavior regarding the 'Host' header of the proxy requests
+is that the proxy ignores the value set in the incoming request. This
+can be changed individually for each route in one of the following ways:
+
+1. using the `preserveHost` filter, that sets the proxy request's 'Host'
+header to the value found in the incoming request object.
+
+2. using the `requestHeader` or a custom filter to set the 'Host' header
+to any arbitrary value. In this case, the header needs to be set in the
+http.Request.Header field and not in the http.Request.Host field.
```
Comment (aryszka) [inline proxy/doc.go]: I wasn't sure if this should be made explicit in the FilterContext by creating a SetHost(string) method.

Comment (danpersa) [inline proxy/doc.go]: I guess it would be more clear to provide an explicit `SetHost` method

**Comment by danpersa** [comment]
lgtm

File: proxy/doc.go (hunk 1 above)
Comment (aryszka) [inline proxy/doc.go]: ok

File: proxy/proxy.go [hunk 2]
```diff
@@ -254,11 +286,16 @@ func (c *filterContext) StateBag() map[string]interface{}    { return c.stateBag
 func (c *filterContext) BackendUrl() string                  { return c.backendUrl }
 func (c *filterContext) OriginalRequest() *http.Request      { return c.originalRequest }
 func (c *filterContext) OriginalResponse() *http.Response    { return c.originalResponse }
+
 func (c *filterContext) Serve(res *http.Response) {
 	c.servedWithResponse = true
 	c.res = res
 }
 
+func (c *filterContext) SetHost(h string) {
```
Comment (lmineiro) [inline proxy/proxy.go]: So, we'll have two functions doing the same? `SetOutgoingHost` and `SetHost` ?

Comment (aryszka) [inline proxy/proxy.go]: no-no, just pushed now to discuss the preserveHost feature. cleanup and doc update is still missing, sorry. i'll make note when ready again for full review.

**Comment by aryszka** [comment]
now it should be ready

**Comment by aryszka** [comment]
now it should be really ready

**Comment by lmineiro** [comment]
lgtm

**Pull request merged**